### PR TITLE
feat(ids-api): Get user profile data from national registry v3.

### PR DESCRIPTION
**Pull request by valurefugl**
https://www.notion.so/Migrate-to-National-Registry-API-V3-fb7b1208972a4dc0b15f49725a63d096?pvs=4

## What

Enable using version 3 of the national registry service when fetching data for user claims.

## Why

To improve performance.

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Integrated National Registry B2C for enhanced crossroad communication and authentication.
  - Added configurations for National Registry B2C in development, production, and staging environments.

- **Enhancements**
  - Updated user profile service to utilize the latest National Registry API.
  - Enhanced test coverage for user profile scenarios, including individual and company authentication.

- **Testing**
  - Introduced functions for generating National Registry V3 user data for testing purposes.

- **Configuration**
  - Added environment-specific configurations for National Registry B2C integration.

### CODE REVIEW

File: libs/feature-flags/src/lib/features.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ export enum Features {
   testing = 'do-not-remove-for-testing-only',
   // Integrate auth-api with user-profile-api.
   userProfileClaims = 'shouldAuthApiReturnUserProfileClaims',
+  shouldAuthIdsApiUseNationalRegistryV3 = 'shouldAuthIdsApiUseNationalRegistryV3',
```
[the remark this answers is not in the document]
Comment (valurefugl) [reply libs/feature-flags/src/lib/features.ts:5]: Not created yet.

File: apps/services/auth/ids-api/src/app/user-profile/user-profile.service.ts [hunk 2]
```diff
@@ -106,24 +113,59 @@ export class UserProfileService {
...
+        gender: this.formatGender(individual.kyn?.kynKodi ?? ''),
+        address: this.formatAddressV3(
+          individual.itarupplysingar?.adsetur ??
+            individual.heimilisfang ??
+            null,
+        ),
+        legalDomicile: this.formatAddressV3(individual.heimilisfang ?? null),
+        birthdate: this.formatBirthdate(
+          individual.faedingarstadur?.faedingarDagur ?? undefined,
+        ),
```
[the remark this answers is not in the document]
Comment (valurefugl) [reply apps/services/auth/ids-api/src/app/user-profile/user-profile.service.ts:146]: Can you please elaborate?

File: libs/feature-flags/src/lib/features.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (valurefugl) [reply libs/feature-flags/src/lib/features.ts:5]: Now created and set to Off in all environments.

**Review by baering — APPROVED** [review]

File: apps/services/auth/ids-api/src/app/user-profile/user-profile.service.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (valurefugl) [reply apps/services/auth/ids-api/src/app/user-profile/user-profile.service.ts:146]: Discussed offline, we will keep this as is.

**Comment by andes-it** [comment]
This PR's had a 15-day siesta. Lost its spark, did it? It's making an exit in 5 days unless the stale label disappears.

**Comment by andes-it** [comment]
This PR's had a 15-day siesta. Lost its spark, did it? It's making an exit in 5 days unless the stale label disappears.

**Comment by saevarma** [comment]
@valurefugl this is waiting for Þjóðskrá's info about dedicated IDS user for the v3 service?

**Comment by valurefugl** [comment]
Yes, waiting for Þjóðskrá.

**Comment by andes-it** [comment]
This PR's had a 15-day siesta. Lost its spark, did it? It's making an exit in 5 days unless the stale label disappears.

File: apps/services/auth/ids-api/test/nationalRegistryV3User.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  return {
+    kennitala,
+    nafn,
+    fulltNafn,
+    kyn,
+    faedingarstadur,
+    heimilisfang,
+    itarupplysingar,
+  }
+}
```
[the remark this answers is not in the document]
Comment (valurefugl) [reply apps/services/auth/ids-api/test/nationalRegistryV3User.ts:70]: already annotated.

File: apps/services/auth/ids-api/src/app/user-profile/user-profile.service.ts [hunk 4]
```diff
@@ -200,7 +242,63 @@ export class UserProfileService {
...
+    if (country == null) country = undefined
+
+    return {
+      formatted,
+      streetAddress,
+      locality,
+      postalCode,
+      country,
+    }
+  }
```
[the remark this answers is not in the document]
Comment (valurefugl) [reply apps/services/auth/ids-api/src/app/user-profile/user-profile.service.ts:301]: already explained.

**Review by lodmfjord — APPROVED** [review]
LGTM

**Pull request merged**